From e869e1d34fde5f3bfba47b10b1db073115ee7152 Mon Sep 17 00:00:00 2001
From: "J.W. Jagersma" <jwjagersma@gmail.com>
Date: Wed, 25 Mar 2020 19:56:27 +0100
Subject: [PATCH] djgpp: emit "b" flag for named bss sections

Unlike ELF, named sections such as .bss.* and .gnu.linkonce.b.* have no
special meaning in COFF, therefore they will have the CONTENTS, LOAD
and DATA attributes set.  The result is that these sections take up
space in object files and executables.  These attributes can be cleared
by emitting the "b" flag in the .section directive.

This can probably be added in default_coff_asm_named_section too.

gcc/
2020-03-25  Jan W. Jagersma  <jwjagersma@gmail.com>

	* config/i386/djgpp.c (i386_djgpp_asm_named_section): Emit "b"
	attribute for SECTION_BSS.
---
 gcc/config/i386/djgpp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gcc/config/i386/djgpp.c b/gcc/config/i386/djgpp.c
index ba6c2d4d3a4..88cf1e6506e 100644
--- a/gcc/config/i386/djgpp.c
+++ b/gcc/config/i386/djgpp.c
@@ -36,6 +36,8 @@ i386_djgpp_asm_named_section(const char *name, unsigned int flags,
     *f++ = 'w';
   if (flags & SECTION_CODE)
     *f++ = 'x';
+  if (flags & SECTION_BSS)
+    *f++ = 'b';
 
   /* LTO sections need 1-byte alignment to avoid confusing the
      zlib decompression algorithm with trailing zero pad bytes.  */
-- 
2.26.2

